### PR TITLE
separate migration from util and make its dependencies into libs

### PR DESCRIPTION
**Pull request by Hawstein**
Part of #6418

separate migration from util and make its dependencies into libs:

* snappy
* kvdb
* error
* common

### CODE REVIEW

**Comment by 5chdn** [comment]
This branch has conflicts.

**Comment by Hawstein** [comment]
@5chdn Conflicts resolved.

**Comment by gavofyork** [comment]
doesn't build...

**Comment by 5chdn** [comment]
CI issues

> Job's log exceeded limit of 4194304 bytes.

File: util/error/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "ethcore-error"
+version = "0.1.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+
+[dependencies]
+rlp = { path = "../rlp" }
+ethcore-bigint = { path = "../bigint" }
+snappy = { path = "../snappy" }
+error-chain = "0.11.0-rc.2"
```
Comment (debris) [inline util/error/Cargo.toml]: it would be good to use a single version of `error-chain`. this pr uses 2 different (`0.11` and `0.11-rc2`)

Comment (Hawstein) [reply util/error/Cargo.toml]: Only version `0.11.0-rc.2` is used. I believe `0.11.0` is auto generated in Cargo.lock which will just drop the suffix `rc.2`.

Comment (debris) [reply util/error/Cargo.toml]: u are right!

**Comment by Hawstein** [comment]
@gavofyork @5chdn @debris Now all checks pass.

**Comment by debris** [comment]
I have a couple of grumbles, but I will fix them by myself and reopen the pr

**Pull request closed without merging**